web/server/codechecker_server/api/authentication.py

```diff
@@ -245,11 +245,62 @@ def createLink(self, provider):
                 code_verifier=pkce_verifier
             )
 
-        self.__insertOAuthSession(state=state,
-                                  code_verifier=pkce_verifier,
-                                  provider=provider)
+        self.insertOAuthSession(state=state,
+                                code_verifier=pkce_verifier,
+                                provider=provider)
         return url
 
+    @timeit
+    def validateOAuthTokenSession(self, access_token):
+        """
+        Returns true of false depending
+        if the OAuth token exists
+        """
+
+        access_token_db = None
+        with DBSession(self.__config_db) as session:
+            access_token_db, *_ = \
+                session.query(OAuthToken.access_token) \
+                .filter(OAuthToken.access_token == access_token) \
+                .first()
+        return access_token_db is not None \
+            and access_token_db == access_token
+
+    @timeit
+    def validateOAuthSession(self, state):
```

Comment:
don't add it as a public function. Make it a server internal function. You can also move this db query to the test code.